geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/api/LifecycleListener.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.distributed.internal.membership.gms.api;
+
+import org.apache.geode.distributed.internal.membership.InternalDistributedMember;
+
+public interface LifecycleListener {
+
+  void start(
+      final MemberIdentifier memberID);
+
+  boolean disconnect(Exception exception);
+
+  /**
+   * TODO - it seems like this could really just happen during {@link #start(MemberIdentifier)}
```

Comment:
TODO when? do we need a ticket instead?

Reply:
https://issues.apache.org/jira/browse/GEODE-7464